packages/devtools_app/lib/src/shared/offline_screen.dart

```diff
@@ -0,0 +1,105 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:collection/collection.dart' show IterableExtension;
+import 'package:flutter/material.dart';
+
+import 'common_widgets.dart';
+import 'globals.dart';
+import 'primitives/auto_dispose.dart';
+import 'screen.dart';
+import 'theme.dart';
+
+/// The screen used for displaying offline data (data that does not require a
+/// connected app to explore).
+///
+/// This screen can be displayed both when DevTools is and is not connected to
+/// an app.
+class OfflineScreenBody extends StatefulWidget {
+  const OfflineScreenBody(this.args, this.possibleScreens);
+
+  final OfflineDataArguments args;
+
+  /// All possible screens, both visible and hidden, that DevTools was started
+  /// with.
+  ///
+  /// This will include screens that are only available when connected to an app
+  /// as well as screens that are only available based on the presence of a
+  /// conditional library.
+  ///
+  /// These screens are stored here so that we can import files for all screens,
+  /// regardless of whether an app is connected to DevTools or whether a
+  /// connected app contains the correct conditional library.
+  final List<Screen> possibleScreens;
+
+  @override
+  _OfflineScreenBodyState createState() => _OfflineScreenBodyState();
+}
+
+class _OfflineScreenBodyState extends State<OfflineScreenBody>
+    with AutoDisposeMixin {
+  Screen? _screen;
+
+  @override
+  void initState() {
+    super.initState();
+    _initScreen();
+    addAutoDisposeListener(offlineController.offlineMode, () {
+      if (!offlineController.offlineMode.value) {
+        setState(() {
+          _screen = null;
+        });
+      }
+    });
```

Comment:
the set state call here used to be called upon the "Exit offline mode" button being clicked. Since we no longer define ExitOfflineButton.onPressed in this widget, we set the _screen to null in response to the offlineMode notifier.